pytest_cookies.py

```diff
@@ -50,17 +60,30 @@ def bake(self, extra_context=None, template=None):
         exception = None
         exit_code = 0
         project_dir = None
+        context = None
 
         if template is None:
             template = self._default_template
 
+        context_file = py.path.local(template).join('cookiecutter.json')
+
         try:
+            # Render the context, so that we can store it on the Result
+            context = prompt_for_config(
+                generate_context(
+                    context_file=str(context_file),
+                    extra_context=extra_context,
+                ),
+                no_input=True,
+            )
+
+            # Run cookiecutter to generate a new project
             project_dir = cookiecutter(
                 template,
                 no_input=True,
                 extra_context=extra_context,
```

Comment:
I am not sure what difference it would make, but I chose to pass the complete rendered context here.  You are probably right to let cookiecutter render it again on it's own without overriding every single value.